### PR TITLE
Workload Identity: Federation bundle endpoint

### LINKED ISSUE #44860: SPIFFE Federation: Add `web-spiffe` compatible endpoint to Proxy
As per https://github.com/gravitational/teleport/pull/43348

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/gravitational/teleport/issues/44860
As per https://github.com/gravitational/teleport/pull/43348

changelog: Adds SPIFFE compatible federation bundle endpoint to the Proxy API, allowing other workload identity platforms to federate with the Teleport cluster.

### CODE REVIEW

File: lib/web/spiffe.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+	bundleBytes, err := bundle.Marshal()
+	if err != nil {
+		return nil, trace.Wrap(err, "marshaling bundle")
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write(bundleBytes)
+	if err != nil {
```
Comment (espadolini) [inline lib/web/spiffe.go:89]:
```suggestion
	if _, err = w.Write(bundleBytes); err != nil {
```

File: lib/web/spiffe.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+)
+
+// getSPIFFEBundle returns the SPIFFE-compatible trust bundle which allows other
+// trust domains to federate with this Teleport cluster.
+//
+// Mounted at /webapi/spiffe/bundle.json
+//
+// Must abide by the standard for a "https_web" profile as described in
+// https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Federation.md#5-serving-and-consuming-a-spiffe-bundle-endpoint
+func (h *Handler) getSPIFFEBundle(w http.ResponseWriter, r *http.Request, _ httprouter.Params) (interface{}, error) {
```
Comment (espadolini) [inline lib/web/spiffe.go:40]:
```suggestion
func (h *Handler) getSPIFFEBundle(w http.ResponseWriter, r *http.Request, _ httprouter.Params) (any, error) {
```

File: lib/web/spiffe.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	bundle.SetRefreshHint(5 * time.Minute)
+	// TODO(noah):
+	// For now, we omit the SequenceNumber field. This is only a SHOULD not a
+	// MUST per the spec. To add this, we will add a sequence number to the
+	// cert authority and increment it on every update.
+
+	spiffeCA, err := h.GetAccessPoint().GetCertAuthority(r.Context(), types.CertAuthID{
+		Type:       types.SPIFFECA,
+		DomainName: cn.GetClusterName(),
+	}, false)
```
Comment (espadolini) [inline lib/web/spiffe.go:68]:
```suggestion
	const loadKeysFalse = false
	spiffeCA, err := h.GetAccessPoint().GetCertAuthority(r.Context(), types.CertAuthID{
		Type:       types.SPIFFECA,
		DomainName: cn.GetClusterName(),
	}, loadKeysFalse)
```

File: lib/web/spiffe_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+	gotCACerts := gotBundle.X509Bundle().X509Authorities()
+	require.Len(t, gotCACerts, len(wantCACerts))
+	for i := range gotCACerts {
+		match := slices.ContainsFunc(wantCACerts, func(want *x509.Certificate) bool {
+			got := gotCACerts[i]
+			return got.Equal(want)
+		})
+		require.True(t, match, "certificate not found in wantCACerts")
+	}
```
Comment (espadolini) [inline lib/web/spiffe_test.go:77]:
```suggestion
	require.Len(t, gotBundle.X509Authorities(), len(wantCACerts))
	for _, caCert := range wantCACerts {
		require.True(t, gotBundle.HasX509Certificate(caCert), "certificate not found in bundle")
	}
```

**Review by espadolini — APPROVED** [review]
all nits

File: lib/web/spiffe_test.go (hunk 4 above)
Comment (strideynet) [reply lib/web/spiffe_test.go:77]: Hah - didn't even realise this was an option - nice!

**Review by timothyb89 — APPROVED** [review]

**Pull request merged**